### DISCUSSION TITLE
How to set nodejs http keepAliveTimeout configuration

### DISCUSSION THREAD

**Discussion by Izabrr** [TypedProject/ts-express-decorators#1882] [Q&A] [answered]
Hi,

I need to change the default value of [keepAliveTimeout](https://nodejs.org/api/http.html#serverkeepalivetimeout). I need to do it because the gateway in front of my application has a different default value (60 sec) and it results in `502` errors when the connection is closed by my app and not yet by the gateway.

I use `PlaformExpress` in my project.

Could you help me ? 🙏 

Thanks

**Comment by Romakita** [comment, marked as answer]
@Inject()
app: PlatformApplication


this.app.getApp() => Express app

**Reply by Izabrr** [reply]
Hi again,

I was already looking for a solution in this `this.app.getApp()` but could you tell me where, precisely, I need to set the custom value for `keepAliveTimeout` ?

Thanks

**Reply by Romakita** [reply]
I don’t know. Search keepAliceTimeout expressjs on google ^^

**Reply by Romakita** [reply]
Hum ok. You have to inject http and https server to set keepAliveTimeout. 

@Inject()
httpServer: Http.Server

@Inject()
httpsServer: Https.Server

this.httpServer. keepAliveTimeout

**Reply by Izabrr** [reply]
It works (and yeah, not only with google help 😉)
Thanks for your support 🙏

**Reply by Romakita** [reply]
You're welcome :)
Don't hesitate to mark this discussions as answered!

**Reply by Romakita** [reply]
## 🎉 Are you happy? 

If you appreciated the support, know that it is **free** and is carried out on **personal time** ;)

A support, even a little bit **makes a difference** for me and  **continues to bring you answers**!

[![github](https://img.shields.io/static/v1?label=Github%20sponsor&message=%E2%9D%A4&logo=GitHub&color=%23fe8e86)](https://github.com/sponsors/romakita) [![opencollective](https://img.shields.io/static/v1?label=OpenCollective%20sponsor&message=%E2%9D%A4&logo=OpenCollective&color=%23fe8e86)](https://opencollective.com/tsed)
